File: backend/cmd/models/Cast.go
package models

import "gorm.io/gorm"

type Cast struct {
	gorm.Model
	ArtistID  uint   `json:"artistID"`
	Artist    Artist `json:"artist"`
	MovieID   uint   `json:"movieID"`
	Movie     Movie  `json:"movie"`
	Character string `gorm:"not null;default:'unknown'" json:"character"`
	Order     int64  `json:"order"`
}

File: backend/models/Cast.js
import mongoose from "mongoose";

const Cast = {
  artist: {
    type: [mongoose.Schema.Types.ObjectId],
    ref: "artist",
    required: [true, "artist is required"],
  },

  character: {
    type: String,
    default: "unknown",
    trim: true,
  },

  order: Number,
};

export default Cast;
